src/decoder.rs

```diff
@@ -764,12 +781,15 @@ fn compute_image(components: &[Component],
             return Ok(data[0].clone())
         }
 
-        let mut buffer = vec![0u8; component.size.width as usize * component.size.height as usize];
-        let line_stride = component.block_size.width as usize * 8;
+        let width = component.size.width as usize;
+        let height = component.size.height as usize;
+
+        let mut buffer = vec![0u8; width * height];
+        let line_stride = width * component.dct_scale;
 
-        for y in 0 .. component.size.height as usize {
-            for x in 0 .. component.size.width as usize {
-                buffer[y * component.size.width as usize + x] = data[0][y * line_stride + x];
+        for y in 0 .. width {
+            for x in 0 .. height {
+                buffer[y * width + x] = data[0][y * line_stride + x];
```

Comment:
#125 seems to have been introduced by this line

Reply:
Looks like I swapped `width` and `height` in the `for` loop ranges when extracting those expressions as variables. 🤦‍♂ 

Also, this presumably means there are no single-channel JPEG images in the test suite?

Reply:
Wow, I missed that ! I fixed that too in #126

Reply:
Thanks!

Reply:
The test suite could clearly be more extensive !